temp_cache.rb

```diff
@@ -1,11 +1,27 @@
 # Temp storage for files
 class TempCache
+  FILE_AGE_OUT_IN_DAYS = 10
+
   attr_reader :cache_path
 
   def initialize(cache_path)
     @cache_path = cache_path
   end
 
+  def self.find_or_create(cache_path)
+    new(cache_path).find_or_create
+  end
+
+  def find_or_create
```

Comment:
I like that this exists now in this class but I feel like it's doing a bunch and that's not super clear. On the flip side I do like the find_or_create rails convention and sticking with that feels natural to me